src/test/test-network.sh

```diff
@@ -18,32 +18,14 @@ ECHO="${ECHO:-echo}"
 # Output is prefixed with the name of the script
 myname=$(basename "$0")
 
-# Save the arguments before we destroy them
-# This might not preserve arguments with spaces in them
-ORIGINAL_ARGS=( "$@" )
-
 # We need to find CHUTNEY_PATH, so that we can call the version of this script
 # in chutney/tools with the same arguments. We also need to respect --quiet.
-until [ -z "$1" ]
-do
-  case "$1" in
-    --chutney-path)
-      CHUTNEY_PATH="$2"
-      shift
-    ;;
-    --tor-path)
-      TOR_DIR="$2"
-      shift
-    ;;
-    --quiet)
-      ECHO=true
-    ;;
-    *)
-      # maybe chutney's test-network.sh can handle it
-    ;;
-  esac
-  shift
-done
+CHUTNEY_PATH=$(echo "$@" | awk -F '--chutney-path ' '{sub(" .*","",$2); print $2}')
+TOR_DIR=$(echo "$@" | awk -F '--tor-dir ' '{sub(" .*","",$2); print $2}')
```

Comment:
How do these awk commands handle quoted strings with spaces?

Reply:
They return the first thing that comes after e.g. `--chutney-path ` until the whitespace character, so it this not handle paths with spaces in them properly. We're not losing any functionality here, as the script didn't know how to handle them previously either - further in the script we're suggesting the user to set `CHUTNEY_PATH` and `TOR_DIR` environment variables as workaround.